frontend/src/metabase/query_builder/components/view/NativeQueryPreviewSidebar/NativeQueryPreviewSidebar.tsx

```diff
@@ -28,18 +24,20 @@ const BUTTON_TITLE = {
   json: t`Convert this question to a native query`,
 };
 
-interface NativeQueryPreviewSidebarProps {
-  question: Question;
-  onLoadQuery: () => Promise<NativeQueryForm>;
-}
+export const NativeQueryPreviewSidebar = (): JSX.Element => {
+  const dispatch = useDispatch();
+  const question = checkNotNull(useSelector(getQuestion));
```

Comment:
This is related to the previous FIXME comment
https://github.com/metabase/metabase/pull/40430/files#diff-1aa8bec6e234f41bbda6930620c697c79436bba4244f9b80e1a768a0a469168dL73

Reply:
@nemanjaglumac I think we need to change `useCreateNativeDatasetMutation` to be actually `Query` with `POST` inside. This would remove `useEffect`. Could you try this?

Reply:
Works like a charm! 🔥 
https://github.com/metabase/metabase/pull/40430/commits/aa323a2c8eaad8ad24ae2256f96cf170d7bbe2db

Reply:
Added a demo to the PR description.